projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/exception/SecretStorageNotConfiguredException.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.exception;
+
+/** Exception thrown, when a secret storage has not been configured */
+public class SecretStorageNotConfiguredException extends ExternalSystemError {
```

Comment:
External System error is 503. That does not seem suitable here. 

503 is generally a re-triable error. And often would be re-tried. But as the secrets feature is disbaled re-trying is unnecessary. 

403  Forbidden (means The server understood the request, but is refusing to fulfill it.)
501 Not Implemented

seem more suitable for this case. I'd vote for 501

Reply:
Makes sense. I'll fix it in the upcoming PR for the tests.